webapp/src/container.tsx

```diff
@@ -528,6 +528,44 @@ export class MainMenu extends data.Component<ISettingsProps, {}> {
         this.props.parent.toggleDebugging();
     }
 
+    componentDidUpdate() {
+        const tutorialOptions = this.props.parent.state.tutorialOptions;
+        const inTutorial = !!tutorialOptions && !!tutorialOptions.tutorial;
+        const header = document.getElementById("mainmenu")
+        if (inTutorial && header && !header.classList.contains("shrink")) {
+            const shrinkHeader = this.shouldShrinkItems();
+            if (shrinkHeader) {
+                header.classList.add("shrink");
+            }
+        }
```

Comment:
```suggestion
        const headerContainsShrink = header.classList.contains("shrink");
        if (inTutorial && header && !headerContainsShrink && this.shouldShrinkItems()) {
            header.classList.add("shrink");
        } else if (headerContainsShrink) {
            header.classList.remove("shrink");
        }
```

it looks like the reason for the weird behavior is because it's not cleared in the update here -- I think this would fix it?

Reply:
hahaha I thought that when the render() returned the div, it would only have classes given by className :\ okay I'll come up with a way to fix it -- i think doing it this way makes it flicker but i will try it again